src/core/mac/mac_frame.cpp

```diff
@@ -294,7 +304,7 @@ uint8_t Frame::FindDstPanIdIndex(void) const
 
     VerifyOrExit(IsDstPanIdPresent(), index = kInvalidIndex);
 
-    index = kFcfSize + kDsnSize;
+    index = kFcfSize + IsSequencePresent();
```

Comment:
This indirectly relies on a `bool` being converted to the value `1` (which is technically correct, but makes the code less readable and harder to understand).